encodings/alp/src/alp/mod.rs

```diff
@@ -35,7 +36,7 @@ mod private {
     impl Sealed for f64 {}
 }
 
-pub trait ALPFloat: private::Sealed + Float + Display + 'static {
+pub trait ALPFloat: private::Sealed + Float + Display + 'static + NativePType {
```

Comment:
minor, but style-wise probably nice to keep lifetime at the end

Reply:
actually that 'static bound is not necessary